common/src/main/java/io/netty/util/concurrent/UnorderedThreadPoolEventExecutor.java

```diff
@@ -212,10 +212,26 @@ public void execute(Runnable command) {
     private static final class RunnableScheduledFutureTask<V> extends PromiseTask<V>
             implements RunnableScheduledFuture<V>, ScheduledFuture<V> {
         private final RunnableScheduledFuture<V> future;
+        private final boolean wasCallable;
 
-        RunnableScheduledFutureTask(EventExecutor executor, RunnableScheduledFuture<V> future) {
+        RunnableScheduledFutureTask(EventExecutor executor, RunnableScheduledFuture<V> future, boolean wasCallable) {
             super(executor, future);
             this.future = future;
+            this.wasCallable = wasCallable;
+        }
+
+        @Override
+        V runTask() throws Exception {
+            V result =  super.runTask();
+            if (result == null && wasCallable) {
+                // If this RunnableScheduledFutureTask wraps a RunnableScheduledFuture that wraps a Callable we need
+                // to ensure that we return the correct result by calling future.get().
+                //
+                // See https://github.com/netty/netty/issues/11072
+                assert future.isDone();
+                return future.get();
```

Comment:
How does this interact with tasks that throw exceptions?

Reply:
actually there was a bug... Fixed it :) Thanks for asking